powerauth-webflow/src/main/js/actions/smsAuthActions.js

```diff
@@ -18,17 +18,18 @@ import {dispatchAction, dispatchError} from "../dispatcher/dispatcher";
 
 /**
  * Get operation detail.
+ * @param component Component requesting the action.
  * @returns {Function} Operation detail.
  */
-export function getOperationData() {
+export function getOperationData(component) {
     return function (dispatch) {
         axios.post("./api/auth/operation/detail", {}, {
             headers: {
                 'X-OPERATION-HASH': operationHash,
             }
         }).then((response) => {
             dispatch({
-                type: "SHOW_SCREEN_SMS",
+                type: "SHOW_SCREEN_" + component,
```

Comment:
This seems too weakly typed - is this OK? I would prefer `switch...case` block with explicit options and default fallback for unknown `component` value.

Reply:
Ok, I will fix it.